C++/C.sublime-syntax

```diff
@@ -505,103 +512,151 @@ contexts:
     - match: \b{{identifier}}\b
 
   statement:
-    - match: (?=\S)
-      push:
-        - match: ;
-          scope: punctuation.terminator.c
-          pop: 1
-        - include: expression
+    - match: ;
+      scope: punctuation.terminator.c
+      pop: 1
+    - include: expression
 
   main:
-    - include: preprocessor
     - include: function
-    - include: statement
+    - match: (?=\S)
+      push: statement
 
   prototype:
-    - include: preprocessor.comment
-    - match: \\$
+    - include: comment
+    - include: preprocessor
+
+  preprocessor.line-escape:
+    - match: \\$\n
       scope: punctuation.separator.continuation.c
       push:
         - match: ^
           pop: 1
+  preprocessor.include:
+    - meta_scope: meta.preprocessor.c
+    - match: \"
+      scope: punctuation.definition.string.begin.c
+      push: string.double
```

Comment:
Strings are commonly used in various contexts and thus may deserve a dedicated context. 

Maybe something like:

```suggestion
    - include: string.quoted.double
```

And in "literal" or "constants" section:

```yml
  string.quoted.double:
    - match: \"
      scope: punctuation.definition.string.begin.c
      push: string.quoted.double.body

  string.quoted.double.body:
    - match: \"
      scope: punctuation.definition.string.end.c
      pop: 1
  
```